numbat/src/tokenizer.rs

```diff
@@ -621,7 +630,12 @@ impl Tokenizer {
                     self.advance();
                 }
 
-                if self.peek().map(|c| c == '.').unwrap_or(false) {
+                if self.peek().map(|c| c == '.').unwrap_or(false)
+                    && self
+                        .peek2()
+                        .map(|c| !is_identifier_start(c))
+                        .unwrap_or(true)
```

Comment:
This logic seems non-trivial. It would be great if we could add tokenizer tests to make sure this works correctly in the future.